formula.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * Created by Howerton on 2/23/2016.
+ */
+
+
+/**
+ * Just for testing, pay no attention to me!
+ */
+function no_args_make_compound(){
+    var input = prompt("Enter compound");
+    //var silver = find_element(input);
+   // document.write("Silver total weight: " + silver.atomic_weight + "<br />");
+
+    var component = Compound_Component(input, 5);
+    document.write("Compound total weight: " + component.component_atomic_weight() + "<br />");
+
+}
+
+/**
+ * Stores a full chemical compound constructed from a single input string.
+ * @param input String to parse
+ * @constructor
+ */
+function Compound(input){
+    var self = {};
+    self.components = string_to_compound(input);
+
+    for(var o = 0; o < components.length; o++){
+        document.write(components.element.name);
+    }
+    self.total_atomic_weight = function calculate_total_atomic_weight(){
+        var sum;
+        for(var i = 0; i < self.components.length; i++){
+            sum += self.components[i].component_atomic_weight();
+        }
+    }
+
+
+}
+
+/**
+ * Separates individual elements and their respective quantities from the compound string.
+ * @param parameters String to parse (i.e. NaCl, H2O)
+ * @returns returns a Compound "object".
+ */
+function string_to_compound(parameters){
```

Comment:
Here are some example compounds this needs to be able to handle:
`H2O NaCl NO2 Na2SO4 HNH4 CHCOO a2 3H2O`

Current regex will match if lowercase letter is in beginning. Need to also handle the initial symbol being a number which is how many compounds there are.